Unskip `html/interaction/focus` WPT tests

Ahmad-S792 opened the pull request:
#### 1d5509d81ba8cfb4a2583c53833209214a77dd3e
<pre>
Unskip `html/interaction/focus` WPT tests

<a href="https://bugs.webkit.org/show_bug.cgi?id=284714">https://bugs.webkit.org/show_bug.cgi?id=284714</a>
<a href="https://rdar.apple.com/141511153">rdar://141511153</a>

Reviewed by Tim Nguyen.

These tests were skipped despite majority of them are passing on WPT, only test which
was failing was `focus-tabindex-order.html` and it was timing out without adding `tabsToLink`
value. It is still kept as `Skip` and we will add &apos;local&apos; copy with necessary changes to
track for any regressions.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/focus-01-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/focus-02-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-negative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-order-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-positive-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-zero-expected.txt:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/287954@main">https://commits.webkit.org/287954@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/796c218365777bdae1f0419c4762dd5093fc816a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81374 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/899 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35317 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85903 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32360 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/917 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/8717 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63514 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21268 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84443 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/623 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74057 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43809 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/518 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28221 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30818 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72006 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28805 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87338 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8604 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6115 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/71826 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/8785 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69878 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71062 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17708 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15122 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14028 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8566 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8402 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/11923 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10210 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-order.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!DOCTYPE html><!-- webkit-test-runner [ TabsToLinks=true ] -->
```
Please do not add those to WPT.

Ahmad-S792 replied:
It is needed to make it pass and also similar is used in `media` (track) related WPT tests.

nt1m replied:
These are generally not exportable to WPT, and as a result, these annotations will generally be lost at the next import. These don't reflect on wpt.fyi either.

Generally, we deal with this in WPT by adding tabindex="0" to buttons, but  in this case, the test is intentionally testing a missing value or an empty value.

I'd rather have this test skipped, since it's not testing default shipping behavior.

Ahmad-S792 replied:
@nt1m - tackled it and now it is as we discussed. :-)

Ahmad-S792 committed:
Unskip `html/interaction/focus` WPT tests

https://bugs.webkit.org/show_bug.cgi?id=284714
rdar://141511153

Reviewed by Tim Nguyen.

These tests were skipped despite majority of them are passing on WPT, only test which
was failing was `focus-tabindex-order.html` and it was timing out without adding `tabsToLink`
value. It is still kept as `Skip` and we will add 'local' copy with necessary changes to
track for any regressions.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/focus-01-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/focus-02-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-negative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-order-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-positive-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/interaction/focus/sequential-focus-navigation-and-the-tabindex-attribute/focus-tabindex-zero-expected.txt:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/287954@main
